#!/usr/bin/env python
PACKAGE = "so3_control"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# World Integral Parameters
gen.add("ki_x",           double_t, 0, "World x integral gain", 0.0,  0.0, 0.01)
gen.add("ki_y",           double_t, 0, "World y integral gain", 0.0,  0.0, 0.01)
gen.add("ki_z",           double_t, 0, "World z integral gain", 0.0,  0.0, 0.01)
gen.add("max_pos_int",    double_t, 0, "World max integral",    0.0,  0.0, 4.00)

# Body Integral Parameters
gen.add("kib_x",          double_t, 0, "Body x integral gain",  0.0,  0.0, 0.01)
gen.add("kib_y",          double_t, 0, "Body y integral gain",  0.0,  0.0, 0.01)
gen.add("kib_z",          double_t, 0, "Body z integral gain",  0.0,  0.0, 0.01)
gen.add("max_pos_int_b",  double_t, 0, "Body max integral",     0.0,  0.0, 2.00)

# Low level corrections (not currently implemented since testing was done on a 
# Pixhawk, which doesn't use these parameters)
# gen.add("corrections/kf", double_t, 0, "kf correction",         0.0,  0.0, 0.0)
# gen.add("corrections/r",  double_t, 0, "Roll offset",           0.0,  0.0, 0.0)
# gen.add("corrections/p",  double_t, 0, "Pitch offset",          0.0,  0.0, 0.0)

gen.add("max_tilt_angle", double_t, 0, "Maximum tilt angle",    0.0,  0.0, 3.14)

exit(gen.generate(PACKAGE, "so3_control", "SO3"))
